src/NServiceBus.Core/Endpoint.cs

```diff
@@ -1,21 +1,59 @@
 namespace NServiceBus
 {
     using System.Threading.Tasks;
+    using ObjectBuilder;
 
     /// <summary>
     /// Provides factory methods for creating and starting endpoint instances.
     /// </summary>
     public static class Endpoint
     {
+        /// <summary>
+        /// Prepares an endpoint to run with and externally managed container.
+        /// </summary>
+        /// <param name="configuration">The endpoint configuration.</param>
+        /// <param name="configureComponents">The registration api adapter for the external container.</param>
+        /// <returns>The prepared endpoint.</returns>
+        public static PreparedEndpoint Prepare(EndpointConfiguration configuration, IConfigureComponents configureComponents)
+        {
+            Guard.AgainstNull(nameof(configuration), configuration);
+            Guard.AgainstNull(nameof(configureComponents), configureComponents);
+
+            configuration.UseExternallyManagedContainer(configureComponents);
+
+            var initializable = configuration.Build();
+
+            return initializable.Prepare();
+        }
+
+        /// <summary>
+        /// Starts a prepared endpoint.
+        /// </summary>
+        /// <param name="preparedEndpoint">The prepared endpoint.</param>
+        /// <param name="builder">The adapter for the containers resolve API.</param>
+        /// <returns>A started endpoint instance.</returns>
+        public static async Task<IEndpointInstance> Start(PreparedEndpoint preparedEndpoint, IBuilder builder)
```

Comment:
it's a bit weird that `Endpoint.Create` returns an `IStartableEndpoint` which is then started via the returned instance (e.g. `startableEndpoint.Start()` to now have a third way to start an endpoint via `Endpoint.Start(Endpoint.Prepare(...))`.

couldn't the prepared endpoint just have a `IStartableEndpoint Initialize(IBuilder builder)` operation instead? It requires an additional step but this isn't supposed to be directly used in most cases anyway?

Reply:
Changing the interface would be a breaking change right? Because it breaks all the folks that implement `IStartableEndpoint`

Reply:
We didn't consider it due to the breaking change. (sounds reasonable in the next major tho)

Reply:
would it make sense to raise an issue for the next major milestone to make sure we won't forget to add it if you think it should be there? Happy to skip this temporarily if this would cause breaking changes.

Reply:
I've added an POA item to raise a follow up issue with API proposal for next major 👍